packages/contracts/contracts/ChugSplashManager.sol

```diff
@@ -349,13 +380,45 @@ contract ChugSplashManager is Owned {
         // completed and reset the active bundle hash so that a new bundle can be executed.
         if (bundle.total == bundle.executions.length) {
             bundle.status = ChugSplashBundleStatus.COMPLETED;
+            bytes32 completedBundleId = activeBundleId;
             activeBundleId = bytes32(0);
 
-            emit ChugSplashBundleCompleted(activeBundleId, msg.sender, bundle.total);
+            emit ChugSplashBundleCompleted(completedBundleId, msg.sender, bundle.total);
             registry.announce("ChugSplashBundleCompleted");
+
+            // Refund the owner's bond
+            (bool success, ) = payable(owner).call{ value: ownerBondAmount }(new bytes(0));
+            require(success, "ChugSplashManager: call to refund owner's bond failed");
+
+            // Refund the executor's bond
+            executorSelectionStrategy.returnExecutorBond(completedBundleId);
         }
     }
 
+    /**
+     * @notice **WARNING**: Cancellation is a potentially dangerous action and should not be
+     *         executed unless in an emergency.
+     *
+     *         Cancels an active ChugSplash bundle. This causes the project owner to forfeit their
+     *         bond, and also refunds the executor's bond.
+     */
+    function cancelActiveChugSplashBundle() public onlyOwner {
+        require(activeBundleId != bytes32(0), "ChugSplashManager: no bundle is currently active");
+
+        ChugSplashBundleState storage bundle = bundles[activeBundleId];
+
+        bytes32 cancelledBundleId = activeBundleId;
+        activeBundleId = bytes32(0);
+        bundle.status = ChugSplashBundleStatus.CANCELLED;
+
+        emit ChugSplashBundleCancelled(cancelledBundleId, msg.sender, bundle.total);
+        registry.announce("ChugSplashBundleCancelled");
+
+        executorSelectionStrategy.returnExecutorBond(cancelledBundleId);
+
+        // TODO: Who should receive the project owner's bond? The executor?
```

Comment:
Executor should receive the bond. We mainly want the bond to prevent the project's owner from trolling executors by starting upgrades, waiting for them to start submitting big transactions, then cancelling and forcing the executors to waste gas on reverting transactions where they can't get paid.